### ISSUE TITLE
Purple "Commercial Break" screen

### ISSUE THREAD

**Issue by uruzrune** [pixeltris/TwitchAdSolutions#257]
Any chance of this being resolved in the near future?

**Comment by stevenya97** [comment]
What browser are you using? Did you apply the uBlock or Userscript scripts? Do you have any other Twitch related extensions installed?

**Comment by uruzrune** [comment]
Firefox 126.0.1 (64-bit).

If I use uBlock Origin only with the usual script:

userResourcesLocation https://raw.githubusercontent.com/pixeltris/TwitchAdSolutions/master/video-swap-new/video-swap-new-ublock-origin.js

It doesn't seem to work at all, and I see ads.

If I turn on TTV LOL PRO, I will no longer see ads, but I will occasionally get the purple break screen.

The only other extension I use that's Twitch related is BetterTTV, and I would not expect something that's emote-focused to affect adblocks.

**Comment by stevenya97** [comment]
Hmm. Let's double check your uBlock Origin dashboard again. Under the `My filters` tab, check/add `twitch.tv##+js(twitch-videoad)`. You are correct chat related extensions shouldn't affect it. Proxies like TTV LOL PRO should just override it but we can ignore that for now since it's not working with it off anyway. 

Let's also check if the script is detecting any ads. If you open the Firefox dev tools(F12), tab over to `Console`, next to filter output make sure `Logs` are highlighted and shown. You should be able to view any stream and if the script finds an ad it should print `Found ads, switch to backup` in the console. There will also be an associated message when the ad is over and the player switches back to the full quality stream. These messages are built into the script itself so it tells us if things are working right. I just tested with a random stream and this behavior does work on my end. Should apply to both prerolls and midrolls. If you do not see these messages, then the script is not being loaded.
![Screenshot 2024-06-03 at 1 25 16 AM](https://github.com/pixeltris/TwitchAdSolutions/assets/10717304/9b6510f3-3ead-49b6-b383-beefb7471ad6)

**Comment by Zevgor** [comment]
Same issue here. Double checked my filters. It has stopped working.

**Comment by Elaws** [comment]
Same issue here - Firefox - Windows 11 - uBlock origin without any scripts.

I followed the advice on the [homepage of TwitchAdSolutions repo](https://github.com/pixeltris/TwitchAdSolutions) to use TTV LOL PRO instead of the scripts.

I very often get these annoying purple screens for 30 seconds.

**Comment by zGato** [comment]
> I very often get these annoying purple screens for 30 seconds.

Read wiki, not an extension-wise issue.

**Comment by Elaws** [comment]
TL;DR : I uninstalled TTV LOL PRO and added the vaft script again. Everything seems to be working fine as it used to. Will report if needed.

> > I very often get these annoying purple screens for 30 seconds.
> 
> Read wiki, not an extension-wise issue.

Thank you @zGato. Not sure what you meant, but by complete luck, I happen to have found [this repo](https://github.com/zGato/ScrewTwitchAds), in which there was a wiki link that seemed to be related to TTV LOL PRO : https://wiki.cdn-perfprod.com/. 

Somewhere in the wiki (in Extension > Troubleshooting > "I get "commercial break in progress" ads), there was a section that seems to describe the problem mentioned in this issue : TL;DR unknown issue, install vaft as workaround / solution.

**Comment by uruzrune** [comment]
@stevenya97 So it doesn't look like the script is being loaded. I've tried it in userResourcesLocation, and I've also tried it under Import on Filter Lists (although I didn't expect that to work, really)

One thing I noticed is that both the vaft script and the standard TwitchAdSolutions script has this as the first line:

"twitch-videoad.js text/javascript"

From what I understand, that's not valid javascript, and it's possible the scripts are just not executing because of this. Can you confirm that line is acceptable?

What else could be preventing uBlock Origin from running this script?

**Comment by uruzrune** [comment]
@stevenya97 so removing the script from uBlock Origin and running it as a userscript through Tampermonkey worked. I had to disable TTV LOL PRO else I would have a black screen with a spinner.

So something is broken with the script execution and uBlock Origin.

**Comment by stevenya97** [comment]
> So something is broken with the script execution and uBlock Origin.

Yeah it is a known issue with uBo, can see #200 for the discussion. uBo removed the `Purge cache` button that would force a reload of the script so there currently isn't a great solution besides using the userscript versions. 

> I had to disable TTV LOL PRO else I would have a black screen with a spinner.

TTV LOL PRO likely conflicts with the script, I would use one or the other for now if you are facing issues.

**Comment by pixeltris** [comment]
Generally you shouldn't get a purple screen with either of the two scripts in this repo. Possibly some geographic locations could cause it but otherwise it's normally caused by the embedded player (though I don't think they use purple screen for it anymore, it's a multi colored thing).
